### PR TITLE
Clinical API in elements

### PR DESCRIPTION
**Pull request by captDaylight**
- adds a public `apolloClinical` to the SDK
- removes the apollo client in the settings page
- updates all Settings queries and mutations to point to SDK apollo instance
- small changes to importing react that was bugging me

### CODE REVIEW

File: packages/sdk/src/lib.ts [hunk 1]
```diff
@@ -95,9 +111,19 @@ export class PhotonClient {
     this.audience = audience;
     this.uri = uri;
     this.clinicalUrl = getClinicalUrl(uri);
+
+    this.clinicalApiUri = `${
+      this.clinicalUrl?.includes('photon')
```
Comment (captDaylight) [inline packages/sdk/src/lib.ts:116]: this I'm looking for feedback on... on the one hand we could pass the clinical uri the way we're passing lambdas... or we could just derive it the way @marisbest2. ain't pretty but it does the job, wdyt?

File: apps/app/src/App.tsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { useColorMode } from '@chakra-ui/react';
 import { BrowserRouter, Route, Routes } from 'react-router-dom';
 
 import { AppState } from '@auth0/auth0-react';
-import { PhotonClient, PhotonProvider } from '@photonhealth/react';
+import { PhotonClient, PhotonProvider } from 'packages/react';
```
Comment (marisbest2) [inline apps/app/src/App.tsx:6]: this seems dangerous / we lose the NX tree I think

File: packages/react/src/provider.tsx [hunk 3]
```diff
@@ -108,6 +114,7 @@ export type GetAllergensReturn = {
 };
 
 export interface PhotonClientContextInterface {
+  clinicalClient: ApolloClient<object> | undefined;
```
Comment (marisbest2) [inline packages/react/src/provider.tsx:117]: this should probably be `ApolloClient<NormalizedCacheObject>`

File: packages/sdk/src/lib.ts [hunk 4]
```diff
@@ -43,8 +43,19 @@ export class PhotonClient {
...
 
+  /**
+   * The GraphQL endpoint of Service's clinical API
+   * */
+  private clinicalApiUri?: string;
+
+  /**
+   * The clinical app url
+   */
   public clinicalUrl?: string;
```
[the remark this answers is not in the document]
Comment (captDaylight) [reply packages/sdk/src/lib.ts:59]: 100% agree

File: packages/elements/src/photon-client/index.tsx [hunk 5]
```diff
@@ -119,29 +119,38 @@ customElement(
...
+
+    createEffect(() => {
       if (!store()?.authentication.state.isLoading) {
         if (!store()?.authentication.state.isAuthenticated && props.autoLogin) {
           const args: any = { appState: {} };
           if (props.redirectPath) {
             args.appState.returnTo = props.redirectPath;
           }
-          await store()?.authentication.login(args);
+          store()?.authentication.login(args);
```
[the remark this answers is not in the document]
Comment (captDaylight) [reply packages/elements/src/photon-client/index.tsx:153]: solid barks if you have a async function handling a reactive effect. The one above I just removed the logic that needs to be async, but in this createEffect case it doesn't need to await on anything

File: apps/app/src/App.tsx (hunk 2 above)
Comment (captDaylight) [reply apps/app/src/App.tsx]: fixed

File: packages/elements/src/photon-client/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (marisbest2) [reply packages/elements/src/photon-client/index.tsx:153]: got it

File: packages/sdk/src/lib.ts (hunk 1 above)
Comment (captDaylight) [reply packages/sdk/src/lib.ts]: check again

File: apps/app/src/App.tsx [hunk 6]
```diff
@@ -24,16 +24,15 @@ import { Prescription } from './views/routes/Prescription';
 import { PrescriptionForm } from './views/routes/PrescriptionForm';
 import { Prescriptions } from './views/routes/Prescriptions';
 import { Settings } from './views/routes/Settings';
-import { ClinicalApiProvider } from './views/routes/Settings/apollo';
 import { Support } from './views/routes/Support';
 import { UpdatePatientForm } from './views/routes/UpdatePatientForm';
+import { Env } from 'packages/sdk/src/utils';
```
Comment (marisbest2) [inline apps/app/src/App.tsx:29]: this should be imported from `@photonhealth/sdk`

File: packages/sdk/src/lib.ts [hunk 7]
```diff
@@ -92,26 +110,36 @@ export class PhotonClient {
       redirect_uri: redirectURI,
       cacheLocation: 'memory'
     });
-    this.audience = audience;
-    this.uri = uri;
-    this.clinicalUrl = getClinicalUrl(uri);
+    this.audience = env ? lambdasApiUrl[env] : audience;
+    this.uri = env ? `${lambdasApiUrl[env]}/graphql` : uri;
+    this.clinicalUrl = env ? clinicalAppUrl[env] : getClinicalUrl(uri);
```
Comment (marisbest2) [inline packages/sdk/src/lib.ts:115]: I think you want the opposite. Especially if `env` has a default value.

**Review by marisbest2 — APPROVED** [review]
Approved with comments

File: apps/app/src/App.tsx (hunk 6 above)
Comment (captDaylight) [reply apps/app/src/App.tsx]: fixed

File: packages/sdk/src/lib.ts (hunk 7 above)
Comment (captDaylight) [reply packages/sdk/src/lib.ts:124]: makes sense, changing

**Pull request merged**